diff -r 73bed5760068 cmake/buildconfig.cmake
--- a/cmake/buildconfig.cmake	Sun Oct 24 19:05:23 2010 +0300
+++ b/cmake/buildconfig.cmake	Sun Oct 24 23:02:21 2010 +0300
@@ -37,6 +37,9 @@
 	set(CMAKE_SHARED_LINKER_FLAGS_RELEASE
 	    "${CMAKE_SHARED_LINKER_FLAGS_RELEASE}    /SUBSYSTEM:WINDOWS /OPT:REF /OPT:ICF")
 
+	set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} /D _SECURE_SCL=0")
+	set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /D _SECURE_SCL=0")
+
 	if (WITH_DYNAMIC_MSVC_RUNTIME)
 	    message(STATUS "Building with dynamic MSVC runtime")
 	    foreach(flag_var
