diff -up firebreath/gen_templates/FireBreathWin.idl.orig firebreath/gen_templates/FireBreathWin.idl
--- firebreath/gen_templates/FireBreathWin.idl.orig	2010-07-27 00:43:21.000000000 +0300
+++ firebreath/gen_templates/FireBreathWin.idl	2010-07-27 00:44:06.000000000 +0300
@@ -75,5 +75,6 @@ library ${FBTYPELIB_NAME}
         [default] interface IFBComJavascriptObject;
         [default, source] dispinterface IFBComEventSource;
     }
+#include "${CMAKE_CURRENT_SOURCE_DIR}/EIDCompatControl.idl"
 
 };
diff -up firebreath/gen_templates/firebreathWin.rc.orig firebreath/gen_templates/firebreathWin.rc
--- firebreath/gen_templates/firebreathWin.rc.orig	2010-07-27 00:43:08.000000000 +0300
+++ firebreath/gen_templates/firebreathWin.rc	2010-07-27 00:15:47.000000000 +0300
@@ -20,6 +20,9 @@
 #include "Win/pininputdialog.rc"
 #include "Win/whitelistdialog_res.h"
 #include "Win/whitelistdialog.rc"
+#ifdef SUPPORT_OLD_APIS
+#include "EIDCompatControl.rc"
+#endif
 
 /////////////////////////////////////////////////////////////////////////////
 #undef APSTUDIO_READONLY_SYMBOLS
@@ -72,8 +72,8 @@
 //
 
 VS_VERSION_INFO VERSIONINFO
- FILEVERSION 1,0,0,1
- PRODUCTVERSION 1,0,0,1
+ FILEVERSION 1,2,0,0
+ PRODUCTVERSION 1,2,0,0
  FILEFLAGSMASK 0x3fL
 #ifdef _DEBUG
  FILEFLAGS 0x1L
@@ -98,7 +98,11 @@
             VALUE "ProductVersion", FBSTRING_PLUGIN_VERSION
             VALUE "FileExtents", FBSTRING_FileExtents
             VALUE "FileOpenName", FBSTRING_PluginName
-            VALUE "MIMEType", FBSTRING_MIMEType
+#ifdef SUPPORT_OLD_APIS
+            VALUE "MIMEType", "application/x-esteid|application/x-digidoc|application/x-idcard-plugin"
+#else
+            VALUE "MIMEType", FBSTRING_MIMEType
+#endif
         END
     END
     BLOCK "VarFileInfo"
