diff -r 69179e87fe85 cmake/buildconfig.cmake
--- a/cmake/buildconfig.cmake	Mon Mar 01 08:16:18 2010 +0100
+++ b/cmake/buildconfig.cmake	Mon Mar 01 11:24:42 2010 +0200
@@ -14,8 +14,8 @@
 
     
 if (WIN32)
-	set(CMAKE_C_FLAGS                            "/DWIN32 /DXP_WIN=1 /W3 /wd4996 /nologo /D UNICODE /D _UNICODE /D _WINDOWS")
-	set(CMAKE_CXX_FLAGS                          "/DWIN32 /DXP_WIN=1 /W3 /wd4996 /nologo /EHsc /wd4290 /D UNICODE /D _UNICODE /D _WINDOWS")
+	set(CMAKE_C_FLAGS                            "/DWIN32 /DXP_WIN=1 /W3 /wd4996 /nologo /D UNICODE /D _UNICODE /D _WINDOWS /D _SECURE_SCL=0")
+	set(CMAKE_CXX_FLAGS                          "/DWIN32 /DXP_WIN=1 /W3 /wd4996 /nologo /EHsc /wd4290 /D UNICODE /D _UNICODE /D _WINDOWS /D _SECURE_SCL=0")
 	set(CMAKE_C_FLAGS_RELEASE                    "/MT /O1 /DNDEBUG")
 	set(CMAKE_CXX_FLAGS_RELEASE                  "/MT /O1 /DNDEBUG")
 	set(CMAKE_C_FLAGS_DEBUG                      "/MTd /Od /DDEBUG /D_DEBUG /ZI /RTC1 /Gm")
