diff --git a/cmake/buildconfig.cmake b/cmake/buildconfig.cmake
--- a/cmake/buildconfig.cmake
+++ b/cmake/buildconfig.cmake
@@ -14,12 +14,12 @@
 
     
 if (WIN32)
-	set(CMAKE_C_FLAGS                            "/DWIN32 /DXP_WIN=1 /W3 /wd4996 /nologo /D UNICODE /D _UNICODE /D _WINDOWS")
+	set(CMAKE_C_FLAGS                            "/DWIN32 /DXP_WIN=1 /W3 /wd4996 /nologo /D UNICODE /D _UNICODE /D _WINDOWS /D _SECURE_SCL=0")
     if (NOT WITH_SYSTEM_BOOST)
         set(CMAKE_C_FLAGS ${CMAKE_C_FLAGS} /D BOOST_ALL_NO_LIB=1)
         set(CMAKE_CPP_FLAGS ${CMAKE_CPP_FLAGS} /D BOOST_ALL_NO_LIB=1)
     endif()
-	set(CMAKE_CXX_FLAGS                          "/DWIN32 /DXP_WIN=1 /W3 /wd4996 /nologo /EHsc /wd4290 /D UNICODE /D _UNICODE /D _WINDOWS")
+	set(CMAKE_CXX_FLAGS                          "/DWIN32 /DXP_WIN=1 /W3 /wd4996 /nologo /EHsc /wd4290 /D UNICODE /D _UNICODE /D _WINDOWS /D _SECURE_SCL=0")
 	set(CMAKE_C_FLAGS_RELEASE                    "/MT /O1 /DNDEBUG")
 	set(CMAKE_CXX_FLAGS_RELEASE                  "/MT /O1 /DNDEBUG")
 	set(CMAKE_C_FLAGS_DEBUG                      "/MTd /Od /DDEBUG /D_DEBUG /ZI /RTC1 /Gm")
